lib/repeatable/expression/base.rb

```diff
@@ -24,6 +24,29 @@ def to_h
         )
       end
 
+      def union(other)
+        if other.is_a?(Union)
+          other.union(self)
+        else
+          Union.new(self, other)
+        end
+      end
+      alias + union
+
+      def intersection(other)
+        if other.is_a?(Intersection)
+          other.intersection(self)
+        else
+          Intersection.new(self, other)
+        end
+      end
+      alias | intersection
```

Comment:
Totally. I was working from memory on what symbols meant, and got it totally backwards.